lib/discover/local.go

```diff
@@ -112,24 +112,41 @@ func (c *localClient) Error() error {
 	return c.beacon.Error()
 }
 
-func (c *localClient) announcementPkt() Announce {
-	return Announce{
-		ID:         c.myID,
-		Addresses:  c.addrList.AllAddresses(),
-		InstanceID: rand.Int63(),
+// announcementPkt appends the local discovery packet to send to msg. Returns
+// true if the packet should be sent, false if there is nothing useful to
+// send.
+func (c *localClient) announcementPkt(msg []byte) ([]byte, bool) {
+	addrs := c.addrList.AllAddresses()
+	if len(addrs) == 0 {
+		// Nothing to announce
+		return msg, false
 	}
-}
 
-func (c *localClient) sendLocalAnnouncements() {
-	msg := make([]byte, 4)
+	if cap(msg) >= 4 {
+		msg = msg[:4]
+	} else {
+		msg = make([]byte, 4)
+	}
 	binary.BigEndian.PutUint32(msg, Magic)
 
-	var pkt = c.announcementPkt()
+	pkt := Announce{
+		ID:         c.myID,
+		Addresses:  addrs,
+		InstanceID: rand.Int63(),
+	}
 	bs, _ := pkt.Marshal()
```

Comment:
Perhaps not throw away this error, and return err == nil as the boolean?

Reply:
It's impossible for this marshal to fail afaik